andela/mervels-bn-backend
# #168131418 Logout Functionality

#### What does this PR do? 
adds logout functionality 
#### Description of Task to be completed?
When a user hits on the logout endpoint, they should be signed out of the application with their session ended
#### How should this be manually tested?
- Clone the repo
- checkout to the branch  ft-logout-user-168131418 
   and pull the changes
- Run npm install
- add the environment variables to the .env file
- Run npm test to check tests
- Run  npm start: dev for development
- Go to postman and head over to `localhost:3000/api/v1/auth/signout`
- Add token to the header
#### Any background context you want to provide?
- You need to be signed in to logout
- In case of test of **successfully log out fails** you need to flush remove all data in Redis
- This can be accomplished by running `redis-cli` then running the command `flushall` to delete all data then run the tests again.
#### What are the relevant pivotal tracker stories?
[#168131418](https://www.pivotaltracker.com/story/show/168131418)
#### Screenshots (if appropriate)
<img width="1078" alt="Screen Shot 2019-09-12 at 17 47 28" src="https://user-images.githubusercontent.com/18899285/64799641-6b5af300-d585-11e9-8910-2b916920e9ea.png">
i
